cmake/SundialsSetupCompilers.cmake

```diff
@@ -79,26 +79,31 @@ endif()
 if(ENABLE_ALL_WARNINGS)
   message(STATUS "Enabling all compiler warnings")
 
+  # Some warning flags are not supported by all compilers so ignore unknown
+  # flags with -Wno-unknown-warning-option. Ironically, this is not supported by
+  # some compilers.
+  set(WARNING_FLAGS
+      "-Wno-unknown-warning-option -Wall -Wpedantic -Wextra -Wshadow \
+-Wwrite-strings -Wcast-align -Wdisabled-optimization -Wvla -Walloca \
+-Wduplicated-cond -Wduplicated-branches")
```

Comment:
We might be able to add back `-Wredundant-decls` when we update the SuperLU_DIST version in the GitHub CI tests. It would be good to add a TODO here or an issue to revisit this when we make that update.